syntaxes/templ.tmLanguage.json

```diff
@@ -389,65 +386,63 @@
         }
       ]
     },
-    "empty-import-expression":{
-      "match": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*$",
-      "captures": {
-        "0": {
-          "patterns": [
-            {
-              "include": "#import-expression-start"
-            }
-          ]
-        }
-      }
-    },
     "import-expression": {
-      "begin": "@(?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\(.*\\)\\s*{$",
-      "beginCaptures": {
-        "0": {
+      "patterns": [
+        {
+          "name": "import-expression.templ",
+          "begin": "(@)((?:[A-z_][A-z_0-9]*\\.)?[A-z_][A-z_0-9]*\\()",
+          "beginCaptures": {
+            "1": {
+              "name": "keyword.control.go"
+            },
+            "2": {
+              "patterns": [
+                {
+                  "include": "source.go"
+                }
+              ]
+            }
+          },
+          "end": "\\s*$",
           "patterns": [
             {
-              "include": "#import-expression-start"
+              "name": "params.import-expression.templ",
```

Comment:
This first inner pattern matches from the previous opening parenthesis to the closing and provides `punctuation.definition.end.bracket.round.go` to the closing. The function parameters inside are also marked with `source.go`.